### PR TITLE
(BKR-351) added indentation based on test/step nesting level

### PR DESCRIPTION
**Pull request by kevpl**
This will only affect tests where `test_name` or `step` has
been passed a block to execute.

The only exception to that statement is `host.exec`'s
logic, as it's more presentable to nest command output one
level under the command itself.

### CODE REVIEW

**Comment by er0ck** [comment]
:+1:

**Comment by anodelman** [comment]
retest this please

File: lib/beaker/logger.rb [hunk 1]
```diff
@@ -176,6 +181,38 @@ def convert string
       end
     end
 
+    def prefix_log_line line
```
Comment (anodelman) [inline lib/beaker/logger.rb]: Yard docs here.

File: lib/beaker/logger.rb [hunk 2]
```diff
@@ -176,6 +181,38 @@ def convert string
...
+        actual_lines.map! do |actual_line|
+          @line_prefix + actual_line
+        end
+        new_line = actual_lines.join("\n")
+        new_line << "\n" if has_ending_newline
+        new_line
+      end
+    end
+
+    def step_in
```
Comment (anodelman) [inline lib/beaker/logger.rb]: Yard docs!

File: lib/beaker/logger.rb [hunk 3]
```diff
@@ -176,6 +181,38 @@ def convert string
...
+      @line_prefix_length += 2
+      @line_prefix = ' ' * @line_prefix_length
+      # puts "stepped into #{@line_prefix_length}"
+    end
+
+    def step_out
```
Comment (anodelman) [inline lib/beaker/logger.rb]: Yard!  Docs!

:)

**Comment by anodelman** [comment]
This looks good - though it would be cool if lines that wrapped were also correctly nested.  Not a blocker for merge, but might be worth a secondary ticket.

**Comment by kevpl** [comment]
@anodelman this code should deal with lines that wrap correctly.

In my testing (`host.install_package("perf")` on a centos-6 host), it worked against lines that included newlines as well as lines that included carriage returns.  Are you seeing something different in your testing?

**Comment by anodelman** [comment]
@kevpl Just suggesting that if a line wraps it should insert a break and then shift the rest of the line over - so that all the data stays at that shift level.

**Comment by kevpl** [comment]
@anodelman it does insert a break, and it does shift the next line over so the output stays at the same shift level.  Are you seeing something different?

**Comment by anodelman** [comment]
retest this please

**Comment by anodelman** [comment]
Here's an example:

```
Installing Puppet Enterprise
  
  qowk4whltl8cj4p.delivery.puppetlabs.net (centos7-64-1) 13:47:58$ mktemp -dt 2015-09-22_13.47.58.XXXXXX
    /tmp/2015-09-22_13.47.58.AiRe00
  
  qowk4whltl8cj4p.delivery.puppetlabs.net (centos7-64-1) executed in 0.01 seconds
  
  qbqq7pla1spvycx.delivery.puppetlabs.net (solaris11-64-2) 13:47:58$ mktemp -dt 2015-09-22_13.47.58.XXXXXX
    /tmp/2015-09-22_13.47.58.IYaO1d
  
  qbqq7pla1spvycx.delivery.puppetlabs.net (solaris11-64-2) executed in 0.04 seconds
  
  qowk4whltl8cj4p.delivery.puppetlabs.net (centos7-64-1) 13:47:58$ cd /tmp/2015-09-22_13.47.58.AiRe00; curl http://neptune.puppetlabs.lan/archives/releases/3.7.2/puppet-enterprise-3.7.2-el-7-x86_64.tar.gz | gunzip | tar -xvf -
      % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                     Dload  Upload   Total   Spent    Left  Speed
      0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0    puppet-enterprise-3.7.2-el-7-x86_64/VERSION
```

**Comment by anodelman** [comment]
Sorry, the incorrect line breaks I was seeing where inserted by Jenkins.  :rage:

**Pull request merged**